lib/instrumentation/transaction.js

```diff
@@ -153,8 +149,15 @@ Transaction.prototype.createSpan = function (...args) {
     return null
   }
 
+  const span = new Span(this, ...args, opts)
+
+  if (this._builtSpans >= this._agent._conf.transactionMaxSpans) {
```

Comment:
Note: The meaning of `_buildSpans` ends up changing slightly in this new world.  The `_buildSpans` property is _always_ incremented regardless of whether a span will be dropped or not.  To get the number of started spans we subtract the number of dropped spans from built spans.